### PR TITLE
Added test for dir parameter in astro:build:done.

### LINKED ISSUE #3607: 🐛 BUG: Dir parameter in `astro:build:done` hook of integration API does not work correctly on Windows.
### What version of `astro` are you using?

1.0.0-beta.46

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

yarn

### What operating system are you using?

Windows

### Describe the Bug

統合APIで設定できるフックの`astro:build:done`の`dir`パラメータは出力先のディレクトリ情報をURLオブジェクトで使用できますが、Windowsの場合に少し不具合を引き起こすようです。
具体的には

```js
'astro:build:done': async ({ dir }) => {
  console.log(dir.pathname)
  // /C:/~~~ omission ~~~
  console.log(path.join(dir.pathname, 'dir'))
  // C:\C:\~~~ omission ~~~\dir
}
```
となって、正しいパスに変換されません。

これを修正するには、Windowsの`dir`は先頭の`/`を削除することだと思います。
実際にそれでうまくいきました。

私のWindowsに関する認識で間違いがあれば教えてください。
この件についてはこちらでPRを作成する予定ですが、問題ないでしょうか？

### Link to Minimal Reproducible Example

No description as it can be reproduced only in Windows environment.

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by dc7290**
## Changes

~~The dir parameter of the astro:build:done hook, which can be set in the integration API, can be used in a URL object with the directory information of the output destination, but this causes a little problem on Windows.~~
~~This PR fixes this.~~

Added test for dir parameter in astro:build:done.
 
## Testing

We added a temporary integration API and tested the dir parameter values we could get there.

## Docs

~~The documentation is not changed because it is a fix for a problem and does not change existing functionality.~~

Add a note to the documentation regarding the use of the dir parameter in the astro:build:done hook.

https://github.com/withastro/docs/pull/808

### CODE REVIEW

**Comment by dc7290** [comment]
Note:

As per this link, the pathname cannot be obtained properly with a URL object, so it seems necessary to use fileToUrlPath.
https://nodejs.org/api/url.html#urlfileurltopathurl

**Comment by dc7290** [comment]
Once I made the change to resolve the issue in question.

However, as I mentioned in one of my previous comments, using a URL object to return the path may not be sufficient.
I need broad input on this point!

**Comment by natemoo-re** [comment]
Hey @dc7290! Really appreciate you jumping in and opening a PR for this issue. I do think we want the public API for `dir` to remain as the cross-platform URL object rather than a platform-specific `path` string. Modifying `dir.pathname` to be a path string rather than the `URL.pathname` is not the correct fix here.

Taking a step back, I think the solution might be to improve our integration documentation to note that you need to use `fileURLToPath(dir)` _before_ reading from the filesystem.

**Comment by dc7290** [comment]
@natemoo-re 

This PR is in effect only an addition of tests, so please close it if you do not need it.
If you decide that you will need my added tests in the future, please merge them in.
It would be a great pleasure for me.

File: packages/astro/test/static-build-dir.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+			],
+		});
+		await fixture.build();
+	});
+	it('dir takes the URL path to the output directory', async () => {
+		const pathname = url.fileURLToPath(checkDir);
+
+		expect(path.relative('', pathname)).to.be.equal(
+			path.join('test/fixtures/static-build-dir/dist')
+		);
```
Comment (natemoo-re) [inline packages/astro/test/static-build-dir.test.js:30]:
```suggestion
		expect(checkDir.toString()).to.be.equal(new URL('./fixtures/static-build-dir/dist', import.meta.url).toString());
```

File: packages/astro/test/static-build-dir.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import url from 'url';
+import path from 'path';
```
Comment (natemoo-re) [inline packages/astro/test/static-build-dir.test.js:2]:
```suggestion
```

**Review by natemoo-re — COMMENTED** [review]
Thanks! I'm going to remove the `path` logic to simplify the test a bit, but otherwise this is a great test!

File: packages/astro/test/static-build-dir.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+							checkDir = dir;
+						},
+					},
+				},
+			],
+		});
+		await fixture.build();
+	});
+	it('dir takes the URL path to the output directory', async () => {
+		expect(checkDir.toString()).to.be.equal(new URL('./fixtures/static-build-dir/dist', import.meta.url).toString());
```
Comment (natemoo-re) [inline packages/astro/test/static-build-dir.test.js:24]:
```suggestion
		expect(checkDir.toString()).to.be.equal(new URL('../fixtures/static-build-dir/dist', import.meta.url).toString());
```

**Comment by dc7290** [comment]
@natemoo-re 
Thanks for the review!
There was an error with the CI, so we've fixed that a bit!

**Review by natemoo-re — APPROVED** [review]
Approved, looks great!

**Pull request merged**